pkg/syncer/statussyncer.go

```diff
@@ -90,11 +105,37 @@ func (c *Controller) updateStatusInUpstream(ctx context.Context, gvr schema.Grou
 	//       I believe to remember that we had resources where that happened.
 
 	upstreamObj.SetResourceVersion(existing.GetResourceVersion())
+
+	if c.advancedSchedulingEnabled {
+		newUpstream := existing.DeepCopy()
+		statusAnnotationValue, err := json.Marshal(downstreamStatus)
+		if err != nil {
+			return err
+		}
+		newUpstreamAnnotations := newUpstream.GetAnnotations()
+		if newUpstreamAnnotations == nil {
+			newUpstreamAnnotations = make(map[string]string)
+		}
+		newUpstreamAnnotations[workloadv1alpha1.InternalClusterStatusAnnotationPrefix+c.pcluster] = string(statusAnnotationValue)
+		newUpstream.SetAnnotations(newUpstreamAnnotations)
+
+		if reflect.DeepEqual(existing, newUpstream) {
+			klog.V(2).Infof("No need to update the status of resource %s|%s/%s from pcluster namespace %s", c.upstreamClusterName, upstreamNamespace, name, downstreamObj.GetNamespace())
```

Comment:
`V(4)`

Compare convention in https://github.com/kcp-dev/kcp/blob/main/CONTRIBUTING.md:

> - Default log-level is 2.
> - Controllers should generally log (a) one line (not more) non-error progress per item with klog.V(2) (b) actions like create/update/delete via klog.V(3) and (c) skipped actions, i.e. what was not done for reasons via klog.V(4).